tf2_ros/include/tf2_ros/message_filter.h

```diff
@@ -95,14 +99,19 @@ static std::string get_filter_failure_reason_string(
   filter_failure_reasons::FilterFailureReason reason)
 {
   switch (reason) {
-    case filter_failure_reasons::Unknown:
-      return "Unknown";
     case filter_failure_reasons::OutTheBack:
-      return "OutTheBack";
+      return
+        "the timestamp on the message is more than the cache length earlier than the newest data"
+        "in the transform cache";
     case filter_failure_reasons::EmptyFrameID:
-      return "EmptyFrameID";
+      return "the frame id of the message is empty";
+    case filter_failure_reasons::NoTransformFound:
+      return "did not found a valid transform, this usually happen at startup ...";
```

Comment:
```suggestion
      return "did not find a valid transform, this usually happens at startup ...";
```

Reply:
I wonder if there's anything we can do about these "expected" startup warnings... I couldn't think of anything reasonable to do though.